compiler/damlc/daml-visual/src/DA/Daml/Visual.hs

```diff
@@ -41,32 +41,33 @@ import DA.Bazel.Runfiles
 import System.FilePath
 import Safe
 import Control.Monad
+import Control.Monad.State
 
 type IsConsuming = Bool
-type InternalChcName = LF.ChoiceName
 
 data Action = ACreate (LF.Qualified LF.TypeConName)
             | AExercise (LF.Qualified LF.TypeConName) LF.ChoiceName deriving (Eq, Ord, Show )
 
 data ChoiceAndAction = ChoiceAndAction
     { choiceName :: LF.ChoiceName
-    , internalChcName :: InternalChcName -- as we have choices with same name across modules
     , choiceConsuming :: IsConsuming
     , actions :: Set.Set Action
     } deriving (Show)
 
 
 data TemplateChoices = TemplateChoices
-    { template :: LF.Template
-    , modName :: LF.ModuleName
+    { template :: LF.Qualified LF.Template
     , choiceAndActions :: [ChoiceAndAction]
     } deriving (Show)
 
+templateChoiceId :: TemplateChoices -> LF.Qualified LF.TypeConName
```

Comment:
I find this function name misleading. I understand that it returns a "choice id" when it actually returns a "template id". Maybe something like `templateIdFromChoices` would be better.

Reply:
Good point, I’ve renamed it to just `templateId`.